src/transations.rs

```diff
@@ -0,0 +1,1359 @@
+    pub fn satisfy(&mut self) -> Result<(), RevaultError> {
+        if let Err(e) = self.descriptor.satisfy(&mut self.txin, &self.satisfier) {
+            return Err(RevaultError::InputSatisfaction(format!(
+                "Script satisfaction error: {}.",
+                e
+            )));
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::super::scripts::{
+        default_unvault_descriptor, default_vault_descriptor, unvault_cpfp_descriptor,
+    };
+    use super::{
+        RevaultError, RevaultPrevout, RevaultSatisfier, RevaultTransaction, RevaultTxOut,
+        RBF_SEQUENCE,
+    };
+
+    use rand::RngCore;
+    use std::str::FromStr;
+
+    use bitcoin::{OutPoint, PublicKey, SigHash, Transaction, TxIn, TxOut};
+    use miniscript::Descriptor;
+
+    fn get_random_privkey() -> secp256k1::SecretKey {
+        let mut rand_bytes = [0u8; 32];
+        let mut secret_key = Err(secp256k1::Error::InvalidSecretKey);
+
+        while secret_key.is_err() {
+            rand::thread_rng().fill_bytes(&mut rand_bytes);
+            secret_key = secp256k1::SecretKey::from_slice(&rand_bytes);
+        }
+
+        secret_key.unwrap()
+    }
+
+    fn get_participants_sets(
+        secp: &secp256k1::Secp256k1<secp256k1::All>,
+    ) -> (
+        (Vec<secp256k1::SecretKey>, Vec<PublicKey>),
+        (Vec<secp256k1::SecretKey>, Vec<PublicKey>),
+        (Vec<secp256k1::SecretKey>, Vec<PublicKey>),
+    ) {
+        let managers_priv = (0..3)
+            .map(|_| get_random_privkey())
+            .collect::<Vec<secp256k1::SecretKey>>();
+        let managers = managers_priv
+            .iter()
+            .map(|privkey| PublicKey {
+                compressed: true,
+                key: secp256k1::PublicKey::from_secret_key(&secp, &privkey),
+            })
+            .collect::<Vec<PublicKey>>();
+
+        let non_managers_priv = (0..8)
+            .map(|_| get_random_privkey())
+            .collect::<Vec<secp256k1::SecretKey>>();
+        let non_managers = non_managers_priv
+            .iter()
+            .map(|privkey| PublicKey {
+                compressed: true,
+                key: secp256k1::PublicKey::from_secret_key(&secp, &privkey),
+            })
+            .collect::<Vec<PublicKey>>();
+
+        let cosigners_priv = (0..8)
+            .map(|_| get_random_privkey())
+            .collect::<Vec<secp256k1::SecretKey>>();
+        let cosigners = cosigners_priv
+            .iter()
+            .map(|privkey| PublicKey {
+                compressed: true,
+                key: secp256k1::PublicKey::from_secret_key(&secp, &privkey),
+            })
+            .collect::<Vec<PublicKey>>();
+
+        (
+            (managers_priv, managers),
+            (non_managers_priv, non_managers),
+            (cosigners_priv, cosigners),
+        )
+    }
+
+    // Routine for ""signing"" a transaction
+    fn satisfy_transaction_input(
+        secp: &secp256k1::Secp256k1<secp256k1::All>,
```

Comment:
`secp256k1::PublicKey::from_secret_key` and `secp.sign` only needs `secp256::Signing` Trait, I think you can change for:
```rust
 fn satisfy_transaction_input<C: Signing>(
        secp: &secp256k1::Secp256k1<C>,
```

Reply:
ah its in the mod tests, I do not really care

Reply:
Yeah, it was just a helpful quick routine for the tests: we can generalize it later if we ever need.